lib/mapit_api.rb

```diff
@@ -1,9 +1,13 @@
 module MapitApi
   class InvalidPostcodeError < StandardError; end
+  class ValidPostcodeNoPlaces < StandardError; end
 
   def self.location_for_postcode(postcode)
     location_data = Imminence.mapit_api.location_for_postcode(postcode)
     raise InvalidPostcodeError if location_data.nil?
+    if areas = location_data.response.to_hash["areas"]
+      raise ValidPostcodeNoPlaces if areas.empty?
```

Comment:
I do like this approach, but we'll need a similar `rescue_from` in the `places_controller` to catch this and turn it into a non explodey error response.  Like we do for `InvalidPostcodeError` (see: https://github.com/alphagov/imminence/blob/master/app/controllers/places_controller.rb#L19-L21).